sdk/storage/azure-storage-blob/azure/storage/blob/_list_blobs_helper.py

```diff
@@ -226,36 +220,29 @@ class BlobPrefix(ItemPaged, DictMixin):
     """An Iterable of Blob properties.
 
     Returned from walk_blobs when a delimiter is used.
-    Can be thought of as a virtual blob directory.
-
-    :ivar str name: The prefix, or "directory name" of the blob.
-    :ivar str service_endpoint: The service URL.
-    :ivar str prefix: A blob name prefix being used to filter the list.
-    :ivar str marker: The continuation token of the current page of results.
-    :ivar int results_per_page: The maximum number of results retrieved per API call.
-    :ivar str next_marker: The continuation token to retrieve the next page of results.
-    :ivar str location_mode: The location mode being used to list results. The available
-        options include "primary" and "secondary".
-    :ivar current_page: The current page of listed results.
-    :vartype current_page: list(~azure.storage.blob.BlobProperties)
-    :ivar str container: The container that the blobs are listed from.
-    :ivar str delimiter: A delimiting character used for hierarchy listing.
-
-    :param callable command: Function to retrieve the next page of items.
-    :param str prefix: Filters the results to return only blobs whose names
-        begin with the specified prefix.
-    :param int results_per_page: The maximum number of blobs to retrieve per
-        call.
-    :param str marker: An opaque continuation token.
-    :param str delimiter:
-        Used to capture blobs whose names begin with the same substring up to
-        the appearance of the delimiter character. The delimiter may be a single
-        character or a string.
-    :param location_mode: Specifies the location the request should be sent to.
-        This mode only applies for RA-GRS accounts which allow secondary read access.
-        Options include 'primary' or 'secondary'.
-    """
-    def __init__(self, *args, **kwargs):
+    Can be thought of as a virtual blob directory."""
+
+    service_endpoint: str
+    """The service URL."""
+    prefix: str
+    """A blob name prefix being used to filter the list."""
+    marker: str
+    """The continuation token of the current page of results."""
+    results_per_page: int
+    """The maximum number of results retrieved per API call."""
+    location_mode: str
+    """The location mode being used to list results. The available
+    options include "primary" and "secondary"."""
+    current_page: List[BlobProperties]
+    """The current page of listed results."""
+    delimiter: str
+    """A delimiting character used for hierarchy listing."""
+    command: Callable
+    """Function to retrieve the next page of items."""
+    container: str
+    """The name of the container."""
+
+    def __init__(self, *args: Any, **kwargs: Any) -> None:
         super(BlobPrefix, self).__init__(*args, page_iterator_class=BlobPrefixPaged, **kwargs)
```

Comment:
azure\storage\blob\_list_blobs_helper.py:248: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "str")  [assignment]
azure\storage\blob\_list_blobs_helper.py:249: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "int")  [assignment]
azure\storage\blob\_list_blobs_helper.py:250: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "str")  [assignment]
azure\storage\blob\_list_blobs_helper.py:251: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "str")  [assignment]
azure\storage\blob\_list_blobs_helper.py:252: error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "str")  [assignment]

Thanks to having **kwargs, I forgot if we have a solution for this or not 😢

Reply:
Hmm I feel like this is something we should run into a lot, but I don't remember that actually happening and/or the pattern we use.

Thinking about it more, it's just `Any` so it should just work and I'm not sure why its complaining here. It's probably just the `Optional` part that is actually causing the problem. However, all the member variables are typed as `Any` as well so I'm not sure why it's unhappy.

Reply:
I... don't see any more errors? Must have been a WIP error but it doesn't exist anymore so I'll take it.